README.md
# AmityCoin

![alt text](https://github.com/Amity-Core/docs/blob/master/images/100x100.png "Coin Logo")

![alt text](https://github.com/Amity-Core/docs/blob/master/images/AmityCoin-not%2520launched-red.png)

![alt text](https://github.com/Amity-Core/docs/blob/master/images/AmityCoin-passing-blue.png)
![alt text](https://github.com/Amity-Core/docs/blob/master/images/AmitySite-beta-yellow.png)
![alt text](https://github.com/Amity-Core/docs/blob/master/images/PaperWallet-active-brightgreen.png)
![alt text](https://github.com/Amity-Core/docs/blob/master/images/WatterWallet-active-brightgreen.png)
![alt text](https://github.com/Amity-Core/docs/blob/master/images/GUILinux-passing-brightgreen.png)
![alt text](https://github.com/Amity-Core/docs/blob/master/images/GUIWindows-fail-red.png)
![alt text](https://github.com/Amity-Core/docs/blob/master/images/GUIMac-unknown-yellow.png)

***

| COIN | INFORMATION |
|-----|-----:|
| Name | AmityCoin |
| Ticker | AMIT |
| Forked | Bytecoin-BCN |
| Algorithm | CryptoNightV7 |
| Difficulty | LWMA-2 |
| Block Time | 90 seconds |
| Block Reward | 298.01896 |
| Emission Speed | 25 |
| Mature Rate | 15 blocks |
| Decimals | 4 |
| P2P | 37000 |
| RPC | 37001 |
| Premine | 0% |

***

### AMITY INFORMATION
**The word Amity means "Friendly Relations" and that couldn't describe us any better**

**The Coin**

AmityCoin is private, untraceable and unlinkable giving all the more reasons to use it. With a blockchain that cannot be analyzed, this means you can feel safe when spending AmityCoin as nobody can see where it & how much is going. Using a Payment ID when pauing somebody can verify that it came from you and nobody else. AmityCoin cannot be hacked, forged or altered for greedy needs as it works upon itself to change its way of thinking to stop attacks from happening. The only way is to be from future, are you from the future?

**The Technology**

AmityCoin has a few features to its technology as well as being private. With the CryptoNightV7 algorithm, this makes ASIC machines obsolete to AmityCoin so they cannot take over the network hashing rate to "steal all the coins" as the hashing algorithm won't let it. It also implements LWMA-2 which makes sure block times stick to 90 seconds. We also have added security with privaye keys for your wallet allowing you to transport it to an online Web Wallet AmityCoin provides.

**The Business Plan**

The long-term goal for AmityCoin is to be globally accepted, as much as FIAT is, so you could use it to spend in near enough any shop you come across. Over the next few years we will hire teams in different countries and cities to try and get AmityCoin accepted in to their shops. Once we have established a connection with the smaller businesses, we will proress to try and get larger businesses on board to see the benefits of AmityCoin.

**The Future**

The promise of a decentralized way of making payments is possible but only with a team and community that is will to do something to make that happen. We believe being decentralized will solve some of the worlds problems but also create some as some still believe banks are safe and are the way forward, never properly knowing what it is like without one.

**Community Projects**

Once we have fully established ourselfs, AmityCoin will set up fund rasing scheme for its members. This will consist of users uploading a link to their own developed software, music, art or anything else in a chance to earn some considerable donations from AmityCoin and its team. This will allow that user to do things that money would of been an object to do.

***

![alt text](https://github.com/Amity-Core/docs/blob/master/images/amity-roadmap.png "Beta-RoadMap")

*This RoadMap needs re-doing. This is just a rough idea on what we've done and yet to do*

***

# What We Need From You

**Developers**
Bringing more developers will help to achieve the goals in line set for the AmityCoin adaption. We intend to keep ASIC resistant so that AmityCoin stays as durable as planned. Developers of different languages will hopefully improve the coins codebase and make it more adaptable. iOS & Android wallets are in future sights allowing mobile users to check and spend their balance while they're *out and about*.

**Investors**
Investors are the way to success in another way that coins without funds to do anything
go extinct as they simply cannot afford to achieve goals. We hope that someone looks
at our coin and its hopeful future and realise this has potential to become a
incredible part of the Crypto and real life communities. This would help us get listed
on an exchange quicker as well as paying for future projects and giveaways.

**Community Members**
We believe these guys are also a big part of Amity because these are the people that
the coin is made for, normal everyday people. By having a community to thrieve upon
will help with adaption as they can verify its every we claim it to be and more. Our Discord
is the best place to stay active with the forever growing community.

***


### TEAM

*[Jack James](https://github.com/Amity-Core)* - Founder and Developer

*[Sajo8](https://github.com/Sajo8)* - Junior Service Developer

*[Watt3r](https://github.com/Watt3r)* - Service Developer

*[Xandar](https://github.com/Xandar123)* - Service Developer

*Aaron Tomsett* - Patron of Amity

*Achie* - Advisor/Marketing

###### Feature Developers

*[Brandon Lehmann](https://github.com/brandonlehmann)* - Developer of [TurtleCoin](https://turtlecoin.lol/)


***

##### Build AmityCoin

[Build Instructions for AmityCoin](https://github.com/AmityCore/AmityCoin/wiki/Building-AmityCoin). Right now we only have Linux Ubuntu build instructions. Please contribute to the docs with other OS build information. 

##### Forking AmityCoin

[This is the first guide made to fork AmityCoin](https://github.com/Amity-Core/docs/blob/master/fork-amitycoin.md), please make PR's to contribute to our docs


***

### Links

**Website**
https://amitycoin.tk/

**Paper Wallet**
https://wallet.amitycoin.tk/ 

**Watter Wallet**
https://paper.amitycoin.tk/

*Xandars block explorer and web wallet to come*

***

### CONTACT

**Discord** - https://discord.gg/k2jYvBT
Our Discord server will be the best place for you to contact as we try and stay online 24/7.

**GitHub** - https://github.com/Amitycoin
Our GitHub may currently be still under review, we have spoke to GitHub themselves and they're lifting the flag on my account. This is where all future repos will be held for Amitycoin.

![alt text](https://github.com/Amity-Core/docs/blob/master/images/200x200.png "Our Fancy Logo")

***


docs/ReleaseNotes.md
**Release notes 1.0.0**

- CryptoNight V7 implemented
- New API for CryptoNote RPC Wallet
- CryptoNote RPC Wallet
- Aggregate multi-addresses for CryptoNote RPC Wallet
- Fusion transactions
- Transactions pool synchronization
- Wallet synchronization speed increase
- High-level API update
- Full network layer refactoring
- New multithreading library
- Improved console logging
- JSON RPC improvements
- Multisignature with API
- Low level API
- High level API improvements
- Updated block reward scheme
- Dynamic maximum block size limit
- Instant transaction pool notifications
- Transaction priority based on tx fee
- Transactions are returned from tx pools after 24 hours
- Fully refactored simplewallet
- Transaction history for simplewallet
- Reset command for simplewallet
- Faster wallet refresh
- Further optimization in daemon RAM consumption
- Various network health updates
- Config and CryptoNote forking how-to update
- High level API implementation
- CryptoNight hash function optimization
- Improvements for wallet JSON RPC API
- Optimized blockchain storage memory usage
- Various code improvements
- JSON RPC v2.0 compatibility
- JSON RPC over TCP added
- Further optimizations
- Fixed a bug with checkpoints behavior
- UI improvements for daemon
- Various updates from our forks
- OSX support
- Further optimizations
- Simplwallet can set extra for transfers
- Improvements in JSON RPC for wallet
- UX improvements in simplewallet
- Win32 compilation
- Port mapping for daemon using UPnP
- Improvements for daemon JSON RPC API
- Formalization of high level API
- Improvements for wallet JSON RPC API
- JSON RPC for wallet
- fixed bug with blockchain storing
- fixed bug with correct display of transfer's change 
- bug fix in simple wallet
- now wallet is still working when daemon stores blockchain and can't serve clients; 
- no-console option for a daemon; 
- fixed problem with network synchronization; 
- major bug fix in simple wallet. 


src/CryptoNoteConfig.h
// Copyright (c) 2011-2016 The Cryptonote developers
// Copyright (c) 2018 The AmityCoin Developers
// Distributed under the MIT/X11 software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#pragma once

#include <cstdint>
#include <initializer_list>

namespace CryptoNote {
namespace parameters {

const uint64_t DIFFICULTY_TARGET                             = 90; // seconds

const uint64_t CRYPTONOTE_MAX_BLOCK_NUMBER                   = 500000000;
const size_t   CRYPTONOTE_MAX_BLOCK_BLOB_SIZE                = 500000000;
const size_t   CRYPTONOTE_MAX_TX_SIZE                        = 1000000000;
const uint64_t CRYPTONOTE_PUBLIC_ADDRESS_BASE58_PREFIX       = 0x1bf3c9;
const size_t   CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW          = 15;
const uint64_t CRYPTONOTE_BLOCK_FUTURE_TIME_LIMIT            = DIFFICULTY_TARGET * 7;

const size_t   BLOCKCHAIN_TIMESTAMP_CHECK_WINDOW             = 60;

const uint64_t MONEY_SUPPLY                                  = UINT64_C(100000000000000);
const unsigned EMISSION_SPEED_FACTOR                         = 25;

static_assert(EMISSION_SPEED_FACTOR <= 4 * sizeof(uint64_t), "Bad EMISSION_SPEED_FACTOR");

const size_t   CRYPTONOTE_REWARD_BLOCKS_WINDOW               = 298.01896;
const size_t   CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE     = 10000; //size of block (bytes) after which reward for block calculated using block size
const size_t   CRYPTONOTE_COINBASE_BLOB_RESERVED_SIZE        = 600;
const size_t   CRYPTONOTE_DISPLAY_DECIMAL_POINT              = 4;
const uint64_t MINIMUM_FEE                                   = UINT64_C(10);
const uint64_t DEFAULT_DUST_THRESHOLD                        = MINIMUM_FEE;

const uint64_t EXPECTED_NUMBER_OF_BLOCKS_PER_DAY             = 24 * 60 * 60 / DIFFICULTY_TARGET;
const size_t   DIFFICULTY_WINDOW                             = EXPECTED_NUMBER_OF_BLOCKS_PER_DAY; // blocks
const size_t   DIFFICULTY_CUT                                = 60;  // timestamps to cut after sorting
const size_t   DIFFICULTY_LAG                                = 15;

const size_t   MAX_BLOCK_SIZE_INITIAL                        =  20 * 1024;
const uint64_t MAX_BLOCK_SIZE_GROWTH_SPEED_NUMERATOR         = 100 * 1024;
const uint64_t MAX_BLOCK_SIZE_GROWTH_SPEED_DENOMINATOR       = 365 * 24 * 60 * 60 / DIFFICULTY_TARGET;

const uint64_t CRYPTONOTE_LOCKED_TX_ALLOWED_DELTA_BLOCKS     = 1;
const uint64_t CRYPTONOTE_LOCKED_TX_ALLOWED_DELTA_SECONDS    = DIFFICULTY_TARGET * CRYPTONOTE_LOCKED_TX_ALLOWED_DELTA_BLOCKS;

const uint64_t CRYPTONOTE_MEMPOOL_TX_LIVETIME                = 60 * 60 * 24;     //seconds, one day
const uint64_t CRYPTONOTE_MEMPOOL_TX_FROM_ALT_BLOCK_LIVETIME = 60 * 60 * 24 * 7; //seconds, one week
const uint64_t CRYPTONOTE_NUMBER_OF_PERIODS_TO_FORGET_TX_DELETED_FROM_POOL = 7;  // CRYPTONOTE_NUMBER_OF_PERIODS_TO_FORGET_TX_DELETED_FROM_POOL * CRYPTONOTE_MEMPOOL_TX_LIVETIME = time to forget tx

const size_t   FUSION_TX_MAX_SIZE                            = CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE * 30 / 100;
const size_t   FUSION_TX_MIN_INPUT_COUNT                     = 12;
const size_t   FUSION_TX_MIN_IN_OUT_COUNT_RATIO              = 4;

const char     CRYPTONOTE_BLOCKS_FILENAME[]                  = "blocks.dat";
const char     CRYPTONOTE_BLOCKINDEXES_FILENAME[]            = "blockindexes.dat";
const char     CRYPTONOTE_BLOCKSCACHE_FILENAME[]             = "blockscache.dat";
const char     CRYPTONOTE_POOLDATA_FILENAME[]                = "poolstate.bin";
const char     P2P_NET_DATA_FILENAME[]                       = "p2pstate.bin";
const char     CRYPTONOTE_BLOCKCHAIN_INDICES_FILENAME[]      = "blockchainindices.dat";
const char     MINER_CONFIG_FILE_NAME[]                      = "miner_conf.json";
} // parameters

const char     CRYPTONOTE_NAME[]                             = "AmityCoin";
const char     GENESIS_COINBASE_TX_HEX[]                     = "010f01ff000188f3b501029b2e4c0281c0b02e7c53291a94d1d0cbff8883f8024f5142ee494ffbbd0880712101607089b09fd3dbe8ea5902143915915cfdf762e0ebe3afe7260415efe3266577";

const uint8_t  CURRENT_TRANSACTION_VERSION                   =  1;
const uint8_t  BLOCK_MAJOR_VERSION_1                         =  1;
const uint8_t  BLOCK_MINOR_VERSION_0                         =  0;

const size_t   BLOCKS_IDS_SYNCHRONIZING_DEFAULT_COUNT        =  10000;  //by default, blocks ids count in synchronizing
const size_t   BLOCKS_SYNCHRONIZING_DEFAULT_COUNT            =  200;    //by default, blocks count in blocks downloading
const size_t   COMMAND_RPC_GET_BLOCKS_FAST_MAX_COUNT         =  1000;

const int      P2P_DEFAULT_PORT                              = 37000;
const int      RPC_DEFAULT_PORT                              = 37001;

const size_t   P2P_LOCAL_WHITE_PEERLIST_LIMIT                =  1000;
const size_t   P2P_LOCAL_GRAY_PEERLIST_LIMIT                 =  5000;

const size_t   P2P_CONNECTION_MAX_WRITE_BUFFER_SIZE          = 32 * 1024 * 1024; // 32 MB
const uint32_t P2P_DEFAULT_CONNECTIONS_COUNT                 = 8;
const size_t   P2P_DEFAULT_WHITELIST_CONNECTIONS_PERCENT     = 70;
const uint32_t P2P_DEFAULT_HANDSHAKE_INTERVAL                = 60;            // seconds
const uint32_t P2P_DEFAULT_PACKET_MAX_SIZE                   = 50000000;      // 50000000 bytes maximum packet size
const uint32_t P2P_DEFAULT_PEERS_IN_HANDSHAKE                = 250;
const uint32_t P2P_DEFAULT_CONNECTION_TIMEOUT                = 5000;          // 5 seconds
const uint32_t P2P_DEFAULT_PING_CONNECTION_TIMEOUT           = 2000;          // 2 seconds
const uint64_t P2P_DEFAULT_INVOKE_TIMEOUT                    = 60 * 2 * 1000; // 2 minutes
const size_t   P2P_DEFAULT_HANDSHAKE_INVOKE_TIMEOUT          = 5000;          // 5 seconds
const char     P2P_STAT_TRUSTED_PUB_KEY[]                    = "8f80f9a5a434a9f1510d13336228debfee9c918ce505efe225d8c94d045fa115";

const std::initializer_list<const char*> SEED_NODES = {
  "35.242.151.134:37000", // Aaron
  "104.199.228.58:37000", // Aaron
  "82.132.228.247:37000", // Aaron
  "35.189.81.181:37000", // Aaron
  "92.13.205.196:37000", // Jack
  "142.93.28.194:37000", // Watt
};

struct CheckpointData {
  uint32_t height;
  const char* blockId;
};

#ifdef __GNUC__
__attribute__((unused))
#endif

// You may add here other checkpoints using the following format:
// {<block height>, "<block hash>"},
const std::initializer_list<CheckpointData> CHECKPOINTS = {
  //{ 10000, "84b6345731e2702cdaadc6ce5e5238c4ca5ecf48e3447136b2ed829b8a95f3ad" },
};
}

#define ALLOW_DEBUG_COMMANDS
